### PR TITLE
adding program to sort two list numbers

### PR DESCRIPTION
**Pull request by sumeet-shinde**
### **Describe your change:**
I am adding difference_sort.py file in maths folder.
By this program we can sort two list numbers which are not common.

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [*] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [*] This pull request is all my own work -- I have not plagiarized.
* [*] I know that pull requests will not be merged if they fail the automated tests.
* [*] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [*] All new Python files are placed inside an existing directory.
* [*] All filenames are in all lowercase characters with no spaces or dashes.
* [*] All functions and variable names follow Python naming conventions.
* [*] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [*] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by xcodz-dot** [comment]
I think you are new to python programming type hinting.

File: maths/difference_sort.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+def diff(a=int, c=int) -> int:
```
Comment (xcodz-dot) [inline maths/difference_sort.py:1]:
```suggestion
def diff(b: list, d: list) -> list:
```
we must take input as parameters and the type hinting must be like this

File: maths/difference_sort.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    7
+    36
+    78
+    96
+    99
+    """
+    # Taking input lists
+    b = input("Enter 1st list numbers: ").split()
+    d = input("Enter 2nd list numbers: ").split()
+    # Making set of list
```
Comment (xcodz-dot) [inline maths/difference_sort.py:20]:
```suggestion
    # Making set of list
```
Algorithms take input as parameters

File: maths/difference_sort.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    x = set(b)
+    y = set(d)
+    # Find numbers which are not same in lists
+    p = y.difference(x)
+    q = x.difference(y)
+    # Make union of 2 lists numbers
+    r = p.union(q)
+    print("Sorted list: ")
+    # Printing numbers in sorted manner
+    print("\n".join(sorted(r, key=int)))
```
Comment (xcodz-dot) [inline maths/difference_sort.py:30]:
```suggestion
    r = p.union(q)
    return list(r)
```

algorithms does not print

File: maths/difference_sort.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+if __name__ == "__main__":
+    diff(5, 5)
```
Comment (xcodz-dot) [inline maths/difference_sort.py:34]:
```suggestion
    diff(5, 5)

```

PEP 8 Error Fix

File: maths/difference_sort.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    Enter 2nd list numbers: 1 2 36 96 78 99
+    Sorted list:
+    3
+    4
+    6
+    7
+    36
+    78
+    96
+    99
```
Comment (xcodz-dot) [inline maths/difference_sort.py:15]:
```suggestion
    Sorting 2 list numbers without same numbers
    
    >>> diff([1, 2, 3, 6, 4, 7], [1, 2, 36, 96, 78, 99])
    [3, 4, 6, 7, 36, 78, 96, 99]
```

Doctest update for my requested changes

File: maths/difference_sort.py (hunk 4 above)
Comment (xcodz-dot) [inline maths/difference_sort.py:34]: You can add input and output code here for testing your function, although doctest will automatically test it

**Review by xcodz-dot — CHANGES_REQUESTED** [review]
These changes are needed for your PR to be merged, feel free to point out any errors in my changes

File: maths/difference_sort.py (hunk 4 above)
Comment (sumeet-shinde) [reply maths/difference_sort.py:34]: Thanks for making changes..

Comment (xcodz-dot) [reply maths/difference_sort.py:34]: Whoops. @sumeet-shinde, the changes i have made have not been accepted by you. Please accept them by clicking a button named `commit suggestion` on every changes i have told you.

File: maths/difference_sort.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+def diff(b: list, d: list) -> list:
+    """
+    Sorting 2 list numbers without same numbers
+
```
Comment (xcodz-dot) [inline maths/difference_sort.py:4]:
```suggestion

```

Remove Trailing Whitespaces

File: maths/difference_sort.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    """
+    # Making set of list
+    x = set(b)
+    y = set(d)
+    # Find numbers which are not same in lists
+    p = y.difference(x)
+    q = x.difference(y)
+    # Make union of 2 lists numbers
+    r = p.union(q)
+    return list(r)
```
Comment (xcodz-dot) [inline maths/difference_sort.py:16]:
```suggestion
    r = list(r)
    r.sort()
    return r
```

**Review by xcodz-dot — CHANGES_REQUESTED** [review]
This is required, lists need to be sorted

File: maths/difference_sort.py (hunk 7 above)
Comment (xcodz-dot) [reply maths/difference_sort.py:15]: @sumeet-shinde, these are a few more changes

**Review by xcodz-dot — APPROVED** [review]
@dhruvmanila @poyea @Kush1101, check this out (Ready to be merged)

**Review by poyea — APPROVED** [review]
I'm not sure if this should go to `sorts/` or `maths/`. It is a kind of set-theoretical trick and makes an assumption of the set of elements. Maybe it's sorts, maybe not.

In the meantime you may add more test cases for this function to demonstrate the contraints, etc.

File: maths/difference_sort.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    p = y.difference(x)
+    q = x.difference(y)
+    # Make union of 2 lists numbers
+    r = p.union(q)
+    r = list(r)
+    r.sort()
+    return r
+
+
+if __name__ == "__main__":
```
Comment (poyea) [inline maths/difference_sort.py:20]:
```suggestion
if __name__ == "__main__":
    import doctest
    doctest.testmod()
```

File: maths/difference_sort.py [hunk 9]
```diff
@@ -0,0 +1,21 @@
+def diff(b: list, d: list) -> list:
```
Comment (poyea) [inline maths/difference_sort.py:1]:
```suggestion
def diff(b: List[int], d: List[int]) -> List[int]:
```

**Review by poyea — CHANGES_REQUESTED** [review]
Please sightly modify the code. Thanks.

**Comment by xcodz-dot** [comment]
@poyea, @sumeet-shinde have made the changes

**Comment by sumeet-shinde** [comment]
Can someone please review my code as early as possible (adding program to sort two list numbers #3671)

**Comment by xcodz-dot** [comment]
@dhruvmanila @poyea, please check this out

**Pull request closed without merging**